.github/pull_request_template.md

```diff
@@ -0,0 +1,9 @@
+Before opening this PR, ensure the following:
+- `./format.sh` was run to apply standard formatting.
+- `make` successfully builds a matching ROM.
+- No new compiler warnings were introduced during the build process.
+    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
```

Comment:
Is there anywhere that mentions to consult in the discord server if it seems like the warnings may be necessary? I suppose people will naturally do this if they struggle with removing certain warnings, but it might be better to just have it written down to let them know that's encouraged. I think it would be better than someone potentially agonising over trying to remove a particularly tough warning.

Reply:
I added it to the contributing.md as a generic note, but not here? Do you think that's fine?
I think in practice people could run into issues at any step, not just the warnings. (e.g. if you don't have matching code)

Reply:
I agree it's not just limited to that one point so the generic statement is a good idea. I suppose what I was looking for in particular was a note or prompt in the PR template that if there is any particular issues preventing a PR they can be discussed, but if you feel as though that's already made clear enough then it's likely fine.